src/amqp-client/connection.cr

```diff
@@ -133,13 +137,19 @@ class AMQP::Client
       end
     end
 
-    def close(msg = "")
+    def close(msg = "", no_wait = true)
       return if @closed
       Log.debug { "Closing connection" }
       write Frame::Connection::Close.new(200_u16, msg, 0_u16, 0_u16)
-      @closed = true
```

Comment:
what about double close? we should probably set `@closed = true` already after `return if @closed`.